pkg/types/tool.go

```diff
@@ -283,6 +284,10 @@ func ParseCredentialArgs(toolName string, input string) (string, string, map[str
 		fields = fields[2:]
 	}
 
+	if alias != "" && !isAlphaNumeric(alias) {
+		return "", "", nil, fmt.Errorf("credential alias must be alphanumeric")
+	}
```

Comment:
I decided to add a constraint that a credential alias needs to be alphanumeric. This is to avoid interfering with credentials for model providers and such.